### PR TITLE
Don't use moveTo() when dealing with uploaded files

### PR DESCRIPTION
**Pull request by jiripudil**
As discussed in #15, `HttpFoundationFactory` calls `UploadedFileInterface::moveTo` which effectively prevents any further manipulation with the original PSR-7 uploaded file. Instead of moving the file to the temporary directory, this patch copies it, and tries to clean up after itself if possible (i.e. if the stream is seekable, which I believe should be true in most of the usual scenarios).

It's still not optimal because it has to read the whole uploaded file into the memory. But it should be quite easy to change it to copy it in smaller chunks if a more considerate approach is desired.

Also related to #39

### CODE REVIEW

File: Factory/HttpFoundationFactory.php [hunk 1]
```diff
@@ -104,32 +104,21 @@ private function getFiles(array $uploadedFiles)
...
+        $temporaryPath = $stream->getMetadata('uri');
+
+        if ($test = !\is_string($temporaryPath) || !is_uploaded_file($temporaryPath)) {
             $temporaryPath = $this->getTemporaryPath();
             $psrUploadedFile->moveTo($temporaryPath);
-
-            $clientFileName = $psrUploadedFile->getClientFilename();
-        }
-
-        if (class_exists('Symfony\Component\HttpFoundation\HeaderUtils')) {
```
Comment (stof) [inline Factory/HttpFoundationFactory.php]: This looks like a different cleanup, not something related to the `moveTo` refactoring. It should be submitted as a dedicated commit (and probably even PR)

**Comment by nicolas-grekas** [comment]
I rebased the PR and changed the implementation to not use `moveTo()` when a file upload is found.
@stof OK for you?

**Comment by nicolas-grekas** [comment]
PR resetted to the originally submitted patch.

**Comment by nicolas-grekas** [comment]
Closing in favor of #68, thank you for submitting.
